### PR TITLE
Add media download headers in py-marqo

### PR DESCRIPTION
**Pull request by wanliAlex**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature

* **What is the current behavior?** (You can also link to an open issue here)
we are still using image_download_headers

* **What is the new behavior (if this is a feature change)?**
we now use media_download_headers and make image_download_headers deprecated

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

* **Other information**:

### CODE REVIEW

File: src/marqo/index.py [hunk 1]
```diff
@@ -282,7 +289,8 @@ def search(self, q: Optional[Union[str, dict]] = None, searchable_attributes: Op
             "reRanker": reranker,
             "boost": boost,
             "textQueryPrefix": text_query_prefix,
-            "hybridParameters": hybrid_parameters
+            "hybridParameters": hybrid_parameters,
+            "colbertFields": colbert_fields
```
Comment (farshidz) [inline src/marqo/index.py]: We can't release this field yet

File: src/marqo/models/marqo_index.py [hunk 2]
```diff
@@ -42,6 +42,7 @@ class FieldFeature(str, Enum):
     LexicalSearch = 'lexical_search'
     ScoreModifier = 'score_modifier'
     Filter = 'filter'
+    Colbert = 'colbert'
```
Comment (farshidz) [inline src/marqo/models/marqo_index.py]: See above comment

**Review by farshidz — CHANGES_REQUESTED** [review]

File: src/marqo/index.py (hunk 1 above)
Comment (wanliAlex) [reply src/marqo/index.py]: Ah yet. I messed up my mainline. Fix it now

Comment (wanliAlex) [reply src/marqo/index.py]: done

File: src/marqo/models/marqo_index.py (hunk 2 above)
Comment (wanliAlex) [reply src/marqo/models/marqo_index.py]: done

**Review by farshidz — DISMISSED** [review]

**Review by farshidz — DISMISSED** [review]

File: src/marqo/index.py [hunk 3]
```diff
@@ -495,7 +504,8 @@ def add_documents(
             device: str = None,
             tensor_fields: List[str] = None,
             use_existing_tensors: bool = False,
-            image_download_headers: dict = None,
+            image_download_headers: Optional[dict] = None,
```
Comment (RaynorChavez) [inline src/marqo/index.py:507]: Thanks, this is nice practice

**Review by RaynorChavez — APPROVED** [review]

**Review by papa99do — APPROVED** [review]
LGTM

**Pull request merged**